internal/internal_event_handlers.go

```diff
@@ -54,6 +54,7 @@ import (
 const (
 	queryResultSizeLimit             = 2000000 // 2MB
 	changeVersionSearchAttrSizeLimit = 2048
+	failureReasonPanic               = "Panic"
```

Comment:
I think, at least so far, we want only `WorkflowError` and `NonDeterminismError` to exist

Reply:
Maybe the right way is to rename the metric name?

Reply:
The only two allowed names are
```
		failureReason := "WorkflowError"
		if failWorkflowTask.Cause == enumspb.WORKFLOW_TASK_FAILED_CAUSE_NON_DETERMINISTIC_ERROR {
			failureReason = "NonDeterminismError"
		}
``

But, the fix is not to add a tag here but consolidate these metrics into a single location